components/console/helpers/tree.rst

```diff
@@ -0,0 +1,233 @@
+
+        ╠═ config
+        ║  ╠═ packages
+        ║  ╚═ routes
+        ║    ╠═ framework.yaml
+        ║    ╚═ web_profiler.yaml
+        ╠═ src
+        ║  ╠═ Command
+        ║  ╠═ Controller
+        ║  ║  ╚═ DefaultController.php
+        ║  ╚═ Kernel.php
+        ╚═ templates
+          ╚═ base.html.twig
+
+:method:`Symfony\\Component\\Console\\Helper\\TreeStyle::compact`
+
+    .. code-block:: text
+
+        ├ config
+        │ ├ packages
+        │ └ routes
+        │   ├ framework.yaml
+        │   └ web_profiler.yaml
+        ├ src
+        │ ├ Command
+        │ ├ Controller
+        │ │ └ DefaultController.php
+        │ └ Kernel.php
+        └ templates
+          └ base.html.twig
+
+:method:`Symfony\\Component\\Console\\Helper\\TreeStyle::light`
+
+    .. code-block:: text
+
+        |-- config
+        |   |-- packages
+        |   `-- routes
+        |       |-- framework.yaml
+        |       `-- web_profiler.yaml
+        |-- src
+        |   |-- Command
+        |   |-- Controller
+        |   |   `-- DefaultController.php
+        |   `-- Kernel.php
+        `-- templates
+            `-- base.html.twig
+
+:method:`Symfony\\Component\\Console\\Helper\\TreeStyle::minimal`
+
+    .. code-block:: text
+
+        . config
+        . . packages
+        . . routes
+        .   . framework.yaml
+        .   . web_profiler.yaml
+        . src
+        . . Command
+        . . Controller
+        . . . DefaultController.php
+        . . Kernel.php
+        . templates
+          . base.html.twig
+
+:method:`Symfony\\Component\\Console\\Helper\\TreeStyle::rounded`
+
+    .. code-block:: text
+
+        ├─ config
+        │  ├─ packages
+        │  ╰─ routes
+        │     ├─ framework.yaml
+        │     ╰─ web_profiler.yaml
+        ├─ src
+        │  ├─ Command
+        │  ├─ Controller
+        │  │  ╰─ DefaultController.php
+        │  ╰─ Kernel.php
+        ╰─ templates
+           ╰─ base.html.twig
+
+Making a Custom Tree Style
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can create your own tree style by passing the characters to the constructor
+of the :class:`Symfony\\Component\\Console\\Helper\\TreeStyle` class::
+
+    use Symfony\Component\Console\Helper\TreeHelper;
+    use Symfony\Component\Console\Helper\TreeStyle;
+
+    $customStyle = new TreeStyle('🟣 ', '🟠 ', '🔵 ', '🟢 ', '🔴 ', '🟡 ');
+
+    // Pass the custom style to the createTree method
+
+    $tree = TreeHelper::createTree($io, null, [
+        'src' =>  [
+            'Command',
+            'Controller' => [
+                'DefaultController.php',
+            ],
+            'Kernel.php',
+        ],
+        'templates' => [
+            'base.html.twig',
+        ],
+    ], $customStyle);
+
+    $tree->render();
+
+The above code will output the following tree:
+
+.. code-block:: text
+
+    🔵 🟣 🟡 src
+    🔵 🟢 🟣 🟡 Command
+    🔵 🟢 🟣 🟡 Controller
+    🔵 🟢 🟢 🟠 🟡 DefaultController.php
+    🔵 🟢 🟠 🟡 Kernel.php
+    🔵 🟠 🟡 templates
+    🔵 🔴 🟠 🟡 base.html.twig
```

Comment:
Please add a versionadded directive below, like for other new features, thanks

Reply:
The whole page is about the new tree helper, should i put the versionadded at the very start of the page ?

Reply:
I think yes, and add it also in `console/style.rst` after the new methods

Reply:
Yes please, so that a reader knows, since when this can be used